node/discover.go

```diff
@@ -16,25 +15,11 @@ import (
 	"github.com/libp2p/go-libp2p/core/peer"
 )
 
+// peerDiscoveryStreamHandler sends a list of peer addresses on the stream. This
+// implements the receiving side of ProtocolIDDiscover.
 func (n *Node) peerDiscoveryStreamHandler(s network.Stream) {
 	defer s.Close()
 
-	s.SetReadDeadline(time.Now().Add(time.Second))
-
-	buf := make([]byte, len(discoverPeersMsg))
-	nr, err := s.Read(buf)
-	if err != nil && !errors.Is(err, io.EOF) {
-		n.log.Warn("failed to read peer discovery request", "error", err)
-		return
-	}
-	if nr == 0 { // they hung up
-		return
-	}
-	if string(buf) != discoverPeersMsg {
```

Comment:
This protocol was an early text based version.  The initiator need not send some magic text message to start it off.  The stream is specific to the protocol ID and ready to go.